### PR TITLE
Do not swallow interrupted exception

### PR DESCRIPTION
**Pull request by kokosing**
Do not swallow interrupted exception

### CODE REVIEW

**Comment by kokosing** [comment]
Fixes https://github.com/prestosql/presto/issues/5422

**Review by wendigo — APPROVED** [review]

**Comment by kokosing** [comment]
@wendigo Added one more commit.

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java [hunk 1]
```diff
@@ -52,6 +52,7 @@
 import static io.airlift.concurrent.Threads.daemonThreadsNamed;
 import static io.airlift.units.DataSize.ofBytes;
 import static java.lang.String.format;
+import static java.lang.Thread.currentThread;
```
Comment (findepi) [inline presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java:55]: This is typically not static imported

**Review by findepi — APPROVED** [review]
%

File: presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java (hunk 1 above)
Comment (kokosing) [reply presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java:55]: `Thread.currentThread()` looks like `ImmutableList.toImmutableList()`

Comment (kokosing) [reply presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java:55]: I applied your comment. I see that this is common. I will leave it for another time.

Comment (findepi) [reply presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java]: > `Thread.currentThread()` looks like `ImmutableList.toImmutableList()`

the similarity is deceptive

**Comment by kokosing** [comment]
Failed with: https://github.com/prestosql/presto/issues/5427

**Pull request merged**